distributed/utils.py

```diff
@@ -618,3 +618,18 @@ def shutting_down(l=_shutting_down):
     to early bind this function rather than look it up when calling it,
     since at shutdown module globals may be cleared.
     """
+
+
+def open_port(host=''):
+    """ Return a probably-open port
+
+    There is a chance that this port will be taken by the operating system soon
```

Comment:
Yes, but another process (or even another library function, e.g. imagine setting up logging into a socket) might take it instead of the intended service. The right way to do this is really to let the intended service (e.g. Bokeh) bind on port 0 and then query the actual port. If Bokeh doesn't support this, then it should :-)